src/sql/workbench/parts/queryHistory/electron-browser/queryHistory.contribution.ts

```diff
@@ -7,5 +7,15 @@ import { Registry } from 'vs/platform/registry/common/platform';
 import { IWorkbenchContributionsRegistry, Extensions as WorkbenchExtensions } from 'vs/workbench/common/contributions';
 import { LifecyclePhase } from 'vs/platform/lifecycle/common/lifecycle';
 import { QueryHistoryWorkbenchContribution } from 'sql/workbench/parts/queryHistory/electron-browser/queryHistory';
+import { CommandsRegistry, ICommandService } from 'vs/platform/commands/common/commands';
+import { IQueryHistoryService } from 'sql/platform/queryHistory/common/queryHistoryService';
 
 Registry.as<IWorkbenchContributionsRegistry>(WorkbenchExtensions.Workbench).registerWorkbenchContribution(QueryHistoryWorkbenchContribution, LifecyclePhase.Restored);
+
+CommandsRegistry.registerCommand({
```

Comment:
Won't this show up even if the extension is installed? Have we tested that scenario? Also having this conflict with the extension command is... strange. Is there some cleaner way to manage this?

Reply:
So currently this would just be weird in that it'd show up even if you don't have the extension installed - but executing it would seemingly do nothing. This is because all of the actual implementation is actually in core ADS - the extension literally just runs a command to enable the extension.

I'm moving the command to only be registered once that enable command is called though since yeah - we shouldn't be doing that if the "extension" isn't installed.

In addition you made me realize we had a bug of sorts where I was starting the query collection service even if the extension wasn't enabled. I went ahead and moved that also to only happen if the enable command is called.

Reply:
And re: conflicting with the extension command. The extension never actually defines the command. I just have it defined in the extension.json so that it shows up in the extension gallery as being contributed. It's a bit weird now but that's intentional since the goal here is to move this all to be completely in the extension so it wouldn't be weirdly split up like this.